File: articles.jdbc/src/test/java/pet/db/jdbc/tool/generator/UpdateArticlePayloadTestDataGenerator.java
package pet.db.jdbc.tool.generator;

import lombok.RequiredArgsConstructor;

import org.springframework.stereotype.Component;

import pet.db.jdbc.model.dto.payload.UpdateArticlePayload;
import pet.db.jdbc.model.dto.Article;

import java.util.List;

@Component
@RequiredArgsConstructor
public class UpdateArticlePayloadTestDataGenerator implements TestDataGenerator<UpdateArticlePayload> {

    private final TestDataGenerator<Article> articleTestDataGenerator;;

    @Override
    public UpdateArticlePayload generateSavedData() {
        return generateSavedData(1).getFirst();
    }

    @Override
    public UpdateArticlePayload generateUnsavedData() {
        return generateUnsavedData(1).getFirst();
    }

    @Override
    public List<UpdateArticlePayload> generateSavedData(Integer dataSize) {
        return articleTestDataGenerator.generateSavedData(dataSize).stream()
                .map(this::convertToUpdateArticlePayload)
                .toList();
    }

    @Override
    public List<UpdateArticlePayload> generateUnsavedData(Integer dataSize) {
        return articleTestDataGenerator.generateUnsavedData(dataSize).stream()
                .map(this::convertToUpdateArticlePayload)
                .toList();
    }

    private UpdateArticlePayload convertToUpdateArticlePayload(Article article) {
        return new UpdateArticlePayload(
                article.getTopic(),
                article.getContent());
    }

}

File: src/test/kotlin/pet/articles/test/tool/generator/UpdateArticlePayloadTestDataGenerator.kt
package pet.articles.test.tool.generator

import org.koin.core.annotation.Named
import org.koin.core.annotation.Single

import pet.articles.model.dto.Article
import pet.articles.model.dto.payload.UpdateArticlePayload

class UpdateArticlePayloadTestDataGenerator(
    private val articleTestDataGenerator: TestDataGenerator<Article>
) : TestDataGenerator<UpdateArticlePayload> {

    override fun generateSavedData(dataSize: Int): List<UpdateArticlePayload> =
        articleTestDataGenerator.generateSavedData(dataSize).map(::convertToUpdateArticlePayload)

    override fun generateUnsavedData(dataSize: Int): List<UpdateArticlePayload> =
        articleTestDataGenerator.generateUnsavedData(dataSize).map(::convertToUpdateArticlePayload)

    override fun generateInvalidData(): UpdateArticlePayload =
        convertToUpdateArticlePayload(articleTestDataGenerator.generateInvalidData())

    private fun convertToUpdateArticlePayload(article: Article): UpdateArticlePayload =
        UpdateArticlePayload(
            topic = article.topic,
            content = article.content
        )
}